astropy/io/fits/connect.py

```diff
@@ -162,7 +164,9 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    hdr = fits_to_time(table.header, t) if astropy_native else table.header
```

Comment:
There was an earlier comment by @saimn about trying to minimize the overhead for normal fits usage. One way to do that would be to replace this with:
```
hdr = table.header
if astropy_native:
    from .fitstime import fits_to_time
    hdr = fits_to_time(hdr, t)
```
and then remove the local import above. That way, `fitstime` would only be imported if it could possibly be used.